### PR TITLE
feat(frontend): support xml, yaml format in explain output

### PR DESCRIPTION
**Pull request by lyang24**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

**Please do not leave this empty!**

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
support xml yaml output formats in explain
- How does this PR work? Need a brief introduction for the changed logic (optional)
adding two keywords and two methods to serialize plan output
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/issues/19123

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

Release note
Public Preview: Do note that the structure and format of EXPLAIN FORMAT YAML, XML is considered unstable, and may change overtime.

After this PR, two new [explain option](https://docs.risingwave.com/docs/current/sql-explain/#explain-options) are supported
`FORMAT (XML | YAML)`
For FORMAT XML, it will output xml for logical and physical plans.
For FORMAT YAML, it will output xml for logical and physical plans.
examples:
YAML
```
CREATE TABLE t (v1 int);
    explain (physical, format yaml) create materialized view m1 as SELECT approx_percentile(0.5) WITHIN GROUP (order by v1) from t;
```
YAML result
```
    name: StreamMaterialize
    fields:
      columns:
      - approx_percentile
      pk_columns: []
      pk_conflict: NoCheck
      stream_key: []
    children:
    - name: StreamGlobalApproxPercentile
      fields:
        quantile: 0.5:Float64
        relative_error: 0.01:Float64
      children:
      - name: StreamExchange
        fields:
          dist: Single
        children:
        - name: StreamLocalApproxPercentile
          fields:
            percentile_col: $expr1
            quantile: 0.5:Float64
            relative_error: 0.01:Float64
          children:
          - name: StreamProject
            fields:
              exprs:
              - t.v1::Float64 as $expr1
              - t._row_id
            children:
            - name: StreamTableScan
              fields:
                columns:
                - v1
                - _row_id
                table: t
              children: []
 ```
 XML
 ```
  CREATE TABLE t (v1 int);
    explain (physical, format xml) create materialized view m1 as SELECT approx_percentile(0.5) WITHIN GROUP (order by v1) from t;
 ```
XML RESULT
 ```
 <XmlNode><name>StreamMaterialize</name><fields><columns>approx_percentile</columns><pk_conflict>NoCheck</pk_conflict></fields><children><name>StreamGlobalApproxPercentile</name><fields><quantile>0.5:Float64</quantile><relative_error>0.01:Float64</relative_error></fields><children><name>StreamExchange</name><fields><dist>Single</dist></fields><children><name>StreamLocalApproxPercentile</name><fields><percentile_col>$expr1</percentile_col><quantile>0.5:Float64</quantile><relative_error>0.01:Float64</relative_error></fields><children><name>StreamProject</name><fields><exprs>t.v1::Float64 as $expr1</exprs><exprs>t._row_id</exprs></fields><children><name>StreamTableScan</name><fields><columns>v1</columns><columns>_row_id</columns><table>t</table></fields></children></children></children></children></children></XmlNode>
 ```
</details>

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/mod.rs [hunk 1]
```diff
@@ -677,6 +683,22 @@ impl Explain for PlanRef {
         serde_json::to_string_pretty(&PrettySerde(explain_ir))
             .expect("failed to serialize plan to json")
     }
+
+    /// Explain the plan node and return a xml string.
+    fn explain_to_xml(&self) -> String {
+        let plan = reorganize_elements_id(self.clone());
+        let explain_ir = plan.explain();
+        // serde_xml_rs::to_string(&PrettySerde(explain_ir))
```
Comment (lyang24) [inline src/frontend/src/optimizer/plan_node/mod.rs:691]: looks like there will be name collision on nested fields which will fail xml serialization? is there another way to do this without a new struct like `PrettySerde` or manually serialize?

Comment (lyang24) [reply src/frontend/src/optimizer/plan_node/mod.rs:691]: failed to serialize plan to xml: Writer { source: LastElementNameNotAvailable }
cc @kwannoel  for some insights

Comment (kwannoel) [reply src/frontend/src/optimizer/plan_node/mod.rs:691]: I took a look at the `serde_xml_rs` library. Looks like it hasn't had activity for 2 years, and some bugs may exist. https://stackoverflow.com/questions/76526592/serde-xml-rs-lastelementnamenotavailable-when-serializing.

I suggest using `quick_xml` with the `serialize` feature. You may opt for a later version of `quick_xml`. I tried it locally -- It encounters recursion limit with `0.27` -- the version rw is using.

**Review by kwannoel — APPROVED** [review]

**Comment by kwannoel** [comment]
Thanks for your contribution! 

There are more related issues for `EXPLAIN`: https://github.com/risingwavelabs/risingwave/issues/19361. If you're interested, feel free to contribute for these as well.

**Comment by kwannoel** [comment]
Could you also add a `Release Note`, since this is a new feature.
You can provide an example of usage for `explain (format json)` and `explain (format yaml)`.

You may take a look at the `Release Note` from https://github.com/risingwavelabs/risingwave/pull/19041.

**Comment by lyang24** [comment]
> Could you also add a `Release Note`, since this is a new feature. You can copy the `Release Note` from #19041, and adapt it for yaml and xml formats.

Yes i mirrored the release note with the example pr, do we also need the release note in commit msg?

**Review by lmatz — APPROVED** [review]

**Comment by kwannoel** [comment]
> > Could you also add a `Release Note`, since this is a new feature. You can copy the `Release Note` from #19041, and adapt it for yaml and xml formats.
> 
> Yes i mirrored the release note with the example pr, do we also need the release note in commit msg?

No just PR description is fine.

**Pull request merged**